temba/channels/models.py

```diff
@@ -2013,10 +2013,8 @@ def send_vumi_message(cls, channel, msg, text):
         is_ussd = channel.channel_type in Channel.USSD_CHANNELS
         channel.config['transport_name'] = 'ussd_transport' if is_ussd else 'mtech_ng_smpp_transport'
 
-        if msg.response_to_id:
-            in_reply_to = Msg.objects.get(pk=msg.response_to_id).external_id
-        else:
-            in_reply_to = None
+        in_reply_to = Msg.objects.values_list(
+            'external_id', flat=True).get(pk=msg.response_to_id)
```

Comment:
wow thanks for following this up, it is very interesting. I need the object too so I can update the field I selected so its beneficial in my case. The values_list approach is a bit counterintuitive for me, especially cos' values_list is actually a lazy QS that returns a list of tuples or single values when flattened but you can still use get/filter on anything you haven't selected in the list.